app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,33 @@
+class MicropostsController < ApplicationController
+  before_action :logged_in_user, only: %i(index create destroy)
+  before_action :correct_user, only: :destroy
+
+  def create
+    @micropost = current_user.microposts.build micropost_params
+    if @micropost&.save
+      flash[:success] = t ".micropost_created"
+      redirect_to root_path
+    else
+      #redirect_to root_path
+      @feed_items = current_user.feed.paginate page: params[:page]
+      render "static_pages/home"
+    end
+  end
+
+  def destroy
+    @micropost.destroy
+    flash[:success] = t ".micropost_deleted"
+    redirect_back fallback_location: root_path
+  end
+
+  private
+
+  def micropost_params
+    params.require(:micropost).permit :content, :picture
+  end
+
+  def correct_user
+    @micropost = current_user.microposts.find_by id: params[:id]
+    redirect_to root_path if @micropost.nil?
```

Comment:
you should use `unless @micropost` instead of ` if @micropost.nil?`
Other, I think you should send message to information for user that reason go to root page